pkg/pods/pod_test.go

```diff
@@ -314,6 +327,59 @@ func TestBuildRunitServices(t *testing.T) {
 	Assert(t).AreEqual(string(bytes), string(expected), "Servicebuilder yaml file didn't have expected contents")
 }
 
+func TestInstall(t *testing.T) {
+	fetcher := uri.NewLoggedFetcher(nil)
+	testContext := util.From(runtime.Caller(0))
+
+	currentUser, err := user.Current()
+	Assert(t).IsNil(err, "test setup: couldn't get current user")
+
+	testLocation := testContext.ExpandPath("hoisted-hello_def456.tar.gz")
+
+	launchables := map[manifest.LaunchableID]manifest.LaunchableStanza{
+		"hello": manifest.LaunchableStanza{
+			LaunchableId:   "hello",
+			Location:       testLocation,
+			LaunchableType: "hoist",
+		},
+	}
+
+	builder := manifest.NewBuilder()
+	builder.SetID("hello")
+	builder.SetLaunchables(launchables)
+	builder.SetRunAsUser(currentUser.Name)
+	manifest := builder.GetManifest()
+
+	testPodDir, err := ioutil.TempDir("", "testPodDir")
+	Assert(t).IsNil(err, "Got an unexpected error creating a temp directory")
+	defer os.RemoveAll(testPodDir)
+
+	pod := Pod{
+		Id:      "testPod",
+		path:    testPodDir,
+		logger:  Log.SubLogger(logrus.Fields{"pod": "testPod"}),
+		Fetcher: fetcher,
+	}
+
+	err = pod.Install(manifest, auth.NopVerifier(), artifact.NewRegistry(nil, uri.DefaultFetcher, osversion.DefaultDetector))
+	Assert(t).IsNil(err, "there should not have been an error when installing")
```

Comment:
on travis i get an error here looking up the "hello" user, but i've clearly set the user on line 350 😖